dynamic_analysis_pkg/dynamic_analysis_pkg/__init__.py

```diff
@@ -0,0 +1,157 @@
+        except:
+            raise KeyError('The file [{}] is not present on the simulation results'.format(file_name))
+
+        if file_name == 'BlueConfig':
+            file_content = current_file.raw().read()
+            bc_str = file_content.decode('utf-8')
+            writable_content = bc_str.replace(self.ORIGIN_CIRCUIT_PATH, self.circuit_dir)
+            writable_content = writable_content.replace(str(self.hpc_dir_path), str(self.local_dir))
+
+            with open(file_output_path, 'w') as fd:
+                fd.write(writable_content)
+
+        else:
+            current_file.download(file_name) # moves to home always
+            move(file_name, file_output_path)
+
+        L.info('- [%s] downloaded', file_name)
+
+    def download_report(self):
+        reports = [x for x in self.files_list if x.endswith('.bbp')]
+        if not reports:
+            L.info('No reports were found')
+            return
+        for report in reports:
+            self.download_file_to_storage(report)
+
+
+class FetchMultipleResults:
+    '''
+    Fetch results based on a list of job_ids
+    '''
+    def __init__(self, token, sim_list_ids, base_working_directory=None):
+        self.values = []
+        for sim in tqdm(sim_list_ids):
+            self.values.append(Results(token, sim, base_working_directory))
+
+
+class JobFinder:
+    '''
+    List the jobs launched using Uncicore in the Service Account Piz-Daint
+    '''
+    def __init__(self, token):
+        self.sorted_jobs = None
+        self.sorted_results = None
+        tr = unicore_client.Transport(token)
+        client = unicore_client.Client(tr, UNICORE_ENDPOINT)
+        self.jobs = client.get_jobs()
+
+    def show_list(self, amount_jobs_to_show=1000):
+        full_jobs_list = []
+        for job in tqdm(self.jobs[:amount_jobs_to_show], desc='Fetching jobs'):
+            raw_time = job.properties['submissionTime']
+            full_time = datetime.strptime(raw_time, '%Y-%m-%dT%H:%M:%S%z')
+            date = full_time.strftime('%d-%m-%Y %H:%M')
+            name = job.properties['name']
+            status = job.properties['status']
+            job_id = PurePath(job.properties['_links']['self']['href']).name
+            full_jobs_list.append([name, date, job_id, full_time, status])
+
+        columns = ['name', 'date', 'job_id', 'timestamp', 'status']
+        self.sorted_results = pd.DataFrame(full_jobs_list, columns=columns).sort_values(
+            by=["timestamp"]
+        ).reset_index()
+
+        showing_columns = ['name', 'date', 'status']
+        print(tabulate(self.sorted_results[showing_columns],
+                       headers=showing_columns))
+
+    def find_id_by_index(self):
+        job_selection_index = input('Please select the index that you want to import:')
+        try:
+            selected_index = int(job_selection_index)
+        except:
+            raise ValueError('ERROR: enter correct index')
```

Comment:
Should not you do a while loop where you requery for the correct input instead of raising ?